### PR TITLE
enhance: upgrade edit page to use same tool catalog as chat page

### PR DESCRIPTION
**Pull request by ryanhopperlowe**

### CODE REVIEW

File: components/chat/chatBar/CatalogListBox.tsx [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    const handleKeyDown = (event: React.KeyboardEvent) => {
+      switch (event.key) {
+        case 'Enter':
+        case 'ArrowUp':
+        case 'ArrowDown':
+          break;
+        default:
+          onUncapturedKeyDown?.(event);
+          break;
+      }
```
Comment (thedadams) [inline components/chat/chatBar/CatalogListBox.tsx:83]: I think a comment explaining what is happening here would be good.

File: components/edit/configure.tsx [hunk 2]
```diff
@@ -1,35 +1,34 @@
-import { useCallback, useContext } from 'react';
+import { gatewayTool } from '@/actions/knowledge/util';
```
Comment (thedadams) [inline components/edit/configure.tsx:1]: Seems strange that these imports changed with one very minor code change.

**Review by thedadams — CHANGES_REQUESTED** [review]
Three additional pieces of feedback I have from running this:
1. When I click on "Featured Tools", I get a nice pop up. I would expect to be able to dismiss this popup by click "outside" of it (or somewhere in the app that is not the popup). I am only able to dismiss it by selecting a tool or by hitting `Esc` on my keyboard.
2. When I click "Add Tool via URL", an input modal pops up. When I enter a URL and hit `Enter`, the modal immediately disappears. It should stay visible until it is determined that the URL I entered actually works.
3. There is no validation done when adding a tool via URL. There was validation added in this PR: https://github.com/gptscript-ai/desktop/pull/397. Now, when I add a URL that doesn't exist (like `github.com/thedadams/dne`), I get into a bad state when I reload the chat, as described in this issue: https://github.com/gptscript-ai/desktop/issues/296. The linked PR also fixed this problem.

File: components/edit/configure.tsx (hunk 2 above)
Comment (ryanhopperlowe) [reply components/edit/configure.tsx:1]: Part of this has to do with resolved merge conflicts, and some of the changes are from me fixing Prettier on my machine

**Review by thedadams — CHANGES_REQUESTED** [review]
I see that a comment was added, but it looks like the functionality hasn't changed and still causes regressions.

**Review by thedadams — APPROVED** [review]

**Comment by thedadams** [comment]
Looks good after error is fixed.

**Comment by ryanhopperlowe** [comment]
> Looks good after error is fixed.

@thedadams I found that I was able to fix the error by upgrading the Typescript version, so I opted for that instead :)

File: components/chat/chatBar/CatalogListBox.tsx [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+      name: 'Notion',
+      description: 'Allows the assistant to interact with Notion.',
+      url: 'github.com/gptscript-ai/tools/apis/notion/write',
+      tags: ['notion', 'documentation', 'notes', 'api'],
+      icon: <SiNotion className="text-lg" />,
+    },
+    {
+      name: 'Hubspot',
+      description: 'Allows the assistant to interact with Hubspot.',
+      url: 'github.com/gptscript-ai/tools/apis/hubspot/crm',
```
Comment (drpebcak) [inline components/chat/chatBar/CatalogListBox.tsx:247]: i have already fixed this, but just fyi the split has happened and this is no longer the correct url. its now `github.com/gptscript-ai/tools/apis/hubspot/crm/write`

**Pull request merged**